content/blog/getting-started-with-volume-snapshots-on-a-kubernetes-cluster-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,840 @@
+
+
+
+A volume snapshot is a copy of the data stored in a PV in K8s at a specific point in time. It provides the ability to create snapshot of a PV from stateful applications. Volume snapshot can be used to back up data of a PV, restore a PV from a previous state, or create a new PV from a snapshot. Volume snapshot provides K8s users with a standardized way to copy the contents of a PV at a particular point in time without creating an entirely new volume. This functionality enables, for example, database administrators to backup databases before performing edit or delete modifications.
+
+
+
+Support of volume snapshots in K8s is only available for CSI driver deployed in the cluster. 
+
+### HPE CSI driver for K8s
+
+
+
+The CSI defines a standard interface for container orchestration systems, like K8s, to expose arbitrary block and file storage systems to their containerized workloads. Support for CSI in K8s was introduced as *alpha* in its v1.9 release, and promoted to *beta* in its v1.10 release. Since v1.13 release, the implementation of the CSI has been in *GA* in K8s. With the adoption of CSI, the K8s volume layer becomes truly extensible. Using CSI, 3rd party storage providers, such as HPE,  can write and deploy plugins exposing new storage systems in K8s without ever having to touch the core K8s code. This gives K8s users more options for storage and makes the system more secure and reliable.
+
+
+
+
+A CSI driver for K8s is a plugin that allows K8s to access different types of storage systems, such as *Azure Disks*, *AWS EBS*, and *HPE Storage*, etc. HPE CSI driver for K8s is one of those CSI driver plugins that follows the K8s CSI specification and enables K8s to use various HPE storage systems, such as *Nimble Storage*, *3PAR* and *Primera*. 
+
+
+
+As part of K8s cluster provisioning in HPE GreenLake for Private Cloud Enterprise, HPE CSI driver for K8s has been installed in the cluster. The installation consists of two components, a _controller_ component and a _per-node_ component. 
+
+1. The controller component is deployed as a *Deployment* on any node in the K8s cluster. It implements the CSI Controller service and a list of sidecar containers, such as _external-provisioner_, _external-attacher_, _external-snapshotter_, and _external-resizer_, etc. These controller sidecar containers typically interact with K8s objects, make calls to the driver’s CSI Controller service, manage K8s events and make the appropriate calls to the CSI driver. 
+
+2. The per-node component is deployed on every node in the cluster through a _DaemonSet_. It implements the CSI Node service and the _node-driver-registrar_ sidecar container that registers the CSI driver to kubelet running on every cluster node and being responsible for making the CSI Node service calls. These calls mount and unmount the storage volume from the HPE storage system, making it available to the Pod to consume.
```

Comment:
You lose me in that second sentence. Is there a way you could write it so it's simpler to understand? Maybe multiple, but shorter sentences?

Maybe this is what you mean? "It implements the CSI node service. It also uses the node-driver-registrar sidecar container, which registers the CSI driver to kubelet that runs on every cluster mode and is responsible for making the CSI Node service calls."

Note... sometimes it's really important to use the word "which " as opposed to "that". "Which" is always preceded by a comma in cases like this, because it provides information that enhances the sentence but isn't absolutely necessary for the sentence to stand alone. So, the use of which or that can make a difference in the meaning.

Reply:
Apologize for this confusion !

Is it any better with below rewriting ?

"The per-node component is deployed as a _DaemonSet_ on every node in the cluster. It implements the CSI Node service, together with the _node-driver-registrar_ sidecar container, which registers the CSI driver to kubelet that runs on every cluster node and is responsible for making the CSI Node service calls."